packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.tsx

```diff
@@ -0,0 +1,472 @@
+    PromiseResolved<ReturnType<typeof apiClient.apiRetryInvoice>>
+  >;
+  apiClientOverrides?: Partial<typeof apiClient>;
+};
+
+export const SubscriptionCreate = ({
+  isMobile = false,
+  profile,
+  customer,
+  selectedPlan,
+  refreshSubscriptions,
+  validatorInitialState,
+  createCustomerInitialState,
+  createSubscriptionInitialState,
+  retryInvoiceInitialState,
+  apiClientOverrides = {},
+}: SubscriptionCreateProps) => {
+  const {
+    apiCreateCustomer,
+    apiCreateSubscriptionWithPaymentMethod,
+    apiRetryInvoice,
+  } = useMemo(() => ({ ...apiClient, ...apiClientOverrides }), [
+    apiClient,
+    apiClientOverrides,
+  ]);
+
+  const [createCustomerState, createCustomer, resetCreateCustomer] = useAwait(
+    apiCreateCustomer,
+    {
+      initialState: createCustomerInitialState,
+      rethrowError: true,
+    }
+  );
+
+  const [
+    createSubscriptionState,
+    createSubscription,
+    resetCreateSubscription,
+  ] = useAwait(apiCreateSubscriptionWithPaymentMethod, {
+    initialState: createSubscriptionInitialState,
+    rethrowError: true,
+  });
+
+  const [retryInvoiceState, retryInvoice, resetRetryInvoice] = useAwait(
+    apiRetryInvoice,
+    {
+      initialState: retryInvoiceInitialState,
+      rethrowError: true,
+    }
+  );
+
+  const [submitNonce, refreshSubmitNonce] = useNonce();
+
+  const resetCreateSubscriptionAndNonce = useCallback(async () => {
+    resetCreateSubscription();
+    refreshSubmitNonce();
+  }, [resetCreateSubscription, refreshSubmitNonce]);
+
+  const createSubscriptionAndRefresh = useCallback(
+    async (...args: Parameters<typeof createSubscription>) => {
+      await createSubscription(...args);
+      refreshSubmitNonce();
+    },
+    [createSubscription, refreshSubmitNonce]
+  );
+
+  const onFormMounted = useCallback(
+    () => Amplitude.createSubscriptionMounted(selectedPlan),
+    [selectedPlan]
+  );
+
+  const onFormEngaged = useCallback(
+    () => Amplitude.createSubscriptionEngaged(selectedPlan),
+    [selectedPlan]
+  );
+
+  const [paymentError, setPaymentError] = useState<PaymentError>({
+    type: '',
+    error: false,
+  });
+
+  const inProgress =
+    createCustomerState.pending || createSubscriptionState.pending;
+
+  const isCardError =
+    createSubscriptionState?.error?.code === 'card_declined' ||
+    createSubscriptionState?.error?.code === 'incorrect_cvc';
+
+  // clear any error rendered with `ErrorMessage` on form change
+  const onChange = useCallback(() => {
+    if (paymentError.error) {
+      setPaymentError({ type: '', error: false });
+    } else if (isCardError) {
+      resetCreateSubscriptionAndNonce();
+    }
+  }, [
+    paymentError,
+    setPaymentError,
+    resetCreateSubscriptionAndNonce,
+    isCardError,
+  ]);
+
+  const [retryStatus, setRetryStatus] = useState<RetryStatus>(null);
+
+  const onSubmit: PaymentFormProps['onSubmit'] = useCallback(
+    (params) =>
+      handleSubscriptionPayment({
```

Comment:
This is more verbose than I'd like, but I'm trying to pass along enough context to the `handleSubscriptionPayment` function so that it or its subroutines could possibly be extracted from this component and put in a shared module.